src/modules/bulletproofs/bulletproofs_util.h

```diff
@@ -38,4 +38,17 @@ static void secp256k1_bulletproofs_le64(unsigned char *output, const uint64_t n)
     output[7] = n >> 56;
 }
 
+/* Check if n is power of two*/
+static int secp256k1_is_power_of_two(size_t n) {
+    return n > 0 && (n & (n - 1)) == 0;
+}
+
+/* Compute the log2 of n. If n is not a power of two, it returns the largest
+ * `k` such that 2^k <= n. Assumes n < 2^64. In Bulletproofs, this is bounded
+ * by len of input vectors which can be safely assumed to be less than 2^64.
+*/
+static size_t secp256k1_bulletproofs_pp_log2(size_t n) {
+    return 64 - 1 - secp256k1_clz64_var((uint64_t)n);
+}
```

Comment:
Is there a reason not to take `uint64_t` as an argument directly?

Reply:
Don't know how the conventions are here. Everywhere in the code, this is called with `size_t` argument. Perhaps, casting at the calling sites is better?

Reply:
I think that we should use `size_t` for lengths, unless we seriously expect to encounter lengths in excess of 2^32 and we want to support that.

Reply:
Though we'd then encounter problems with array indexing etc

Reply:
Keeping this as `size_t` for now.